web/components/FcDrawerViewCollisionReports.vue

```diff
@@ -0,0 +1,304 @@
+            <v-menu>
+              <template v-slot:activator="{ on, attrs }">
+                <FcButton
+                  v-bind="attrs"
+                  v-on="on"
+                  class="ml-2"
+                  type="secondary"
+                  :loading="loadingDownload">
+                  <v-icon color="primary" left>mdi-cloud-download</v-icon> Download
+                </FcButton>
+              </template>
+              <v-list>
+                <v-list-item
+                  v-for="{ label, value } in itemsDownloadFormats"
+                  :key="value"
+                  @click="actionDownload(value)">
+                  <v-list-item-title>
+                    {{label}}
+                  </v-list-item-title>
+                </v-list-item>
+              </v-list>
+            </v-menu>
+          </div>
+        </div>
+      </section>
+    </template>
+  </div>
+</template>
+
+<script>
+import { saveAs } from 'file-saver';
+import { mapGetters, mapMutations, mapState } from 'vuex';
+
+import { ReportFormat, ReportType } from '@/lib/Constants';
+import { reporterFetch } from '@/lib/api/BackendClient';
+import {
+  getLocationByFeature,
+} from '@/lib/api/WebApi';
+import FcDialogConfirm from '@/web/components/dialogs/FcDialogConfirm.vue';
+import FcButton from '@/web/components/inputs/FcButton.vue';
+import FcReport from '@/web/components/reports/FcReport.vue';
+import FcMixinRouteAsync from '@/web/mixins/FcMixinRouteAsync';
+
+const DOWNLOAD_FORMATS_SUPPORTED = [
+  ReportFormat.CSV,
+  ReportFormat.PDF,
+];
+
+export default {
+  name: 'FcDrawerViewReports',
+  mixins: [FcMixinRouteAsync],
+  components: {
+    FcButton,
+    FcDialogConfirm,
+    FcReport,
+  },
+  data() {
+    return {
+      indexActiveReportType: 0,
+      leaveConfirmed: false,
+      loadingDownload: false,
+      loadingReportLayout: false,
+      nextRoute: null,
+      reportLayout: null,
+      reportTypes: [
+        ReportType.COLLISION_TABULATION,
+        ReportType.COLLISION_DIRECTORY,
+      ],
+      showConfirmLeave: false,
+    };
+  },
+  computed: {
+    activeReportType() {
+      const { indexActiveReportType, reportTypes } = this;
+      if (indexActiveReportType >= reportTypes.length) {
+        return null;
+      }
+      return reportTypes[indexActiveReportType];
+    },
+    itemsDownloadFormats() {
+      if (this.downloadLoading || this.loadingReportLayout) {
+        return [];
+      }
+      return DOWNLOAD_FORMATS_SUPPORTED
+        .filter(reportFormat => this.activeReportType.formats.includes(reportFormat))
+        .map(({ name }) => ({ label: name, value: name }));
+    },
+    reportParameters() {
+      // TODO: we'll probably have to figure this one out...
+      return {};
+    },
+    ...mapState(['location']),
+    ...mapGetters('viewData', ['filterChipsCollision']),
+  },
+  watch: {
+    activeReportType() {
+      this.updateReportLayout();
+    },
+  },
+  beforeRouteLeave(to, from, next) {
```

Comment:
as this gets closer to complete, we should keep an eye on duplicated code between this and `FcDrawerViewReports` - we can always extract some of that into a mixin.  (I doubt we'll want to fold them into a single component, though, as there's likely to be a lot of small differences in filter, parameter, etc. handling.)